Doc 438 - Add docs for mc support (#478)
minio/docs: source/support/support-overview.rst

=======
Support
=======

.. default-domain:: minio

.. contents:: Table of Contents
   :local:
   :depth: 1

Overview
--------

MinIO users have two options for support.

#. Community support from the `public Slack channel <https://minio.slack.com>`_.
   
   Community support is best-effort only and has no :abbr:`SLA (Service Level Agreement)` or :abbr:`SLO (Service Level Objective)`.

> #. A paid commercial license grants access to the MinIO Subscription Network, `SUBNET <https://subnet.min.io>`_ with a 24 hour or 1 hour :abbr:`SLA (Service Level Agreement)` depending on subscription level.

Reviewer: ```suggestion
#. The MinIO Subscription Network `SUBNET <https://subnet.min.io>`_. SUBNET provides either 24 hour or 1 hour :abbr:`SLA (Service Level Agreement)` depending on subscription level.
```

Author: NIce.

   
   For current licensing levels and pricing, refer to the |SUBNET| page.

.. _minio-docs-subnet:

SUBNET
------

MinIO `SUBNET <https://subnet.min.io?ref=docs>`_ delivers 24/7/365 Direct-to-engineer support through a MinIO-built portal that blends the best of Slack and Zendesk.

Features of SUBNET include:

- Security and architecture reviews (depending on :abbr:`SLA (Service Level Agreement)`)
- Access to Panic Button (depending on :abbr:`SLA (Service Level Agreement)`)
- Secure communication channel to exchange logs and software binaries
- Unlimited seats for your team
- Unlimited issues

> 

Reviewer: We should point to https://min.io/pricing for more information.

Author: Agreed.

Registering Your MinIO Deployment with SUBNET
---------------------------------------------

.. tab-set::

   .. tab-item:: Console

      You can register for SUBNET from the MinIO Console.

      #. Go to your MinIO cluster's URL, then sign in
      #. Select the :guilabel:`Support` option
      #. Select :guilabel:`Register`
      #. Enter your :guilabel:`Email` address and a unique :guilabel:`Password`, then select :guilabel:`Register`

      For clusters without direct Internet access:

      #. Go to your MinIO cluster's URL, then sign in
      #. Select the :guilabel:`Support` option
      #. Select :guilabel:`Register`
      #. Select the :guilabel:`Offline Activation` link at the bottom of the screen

>       #. Copy the registration token for your cluster from item 1

Reviewer: Maybe we can labelize this section and point to it for `mc support register --airgap`

Author: It's Console, not CLI.
I'll see about adding something to the Console docs.

Author: Just added a new tab instead. Console docs don't have any procedures yet, and I'm not ready to open that can of worms.

   :align: center

Reviewing Health Data in SUBNET
-------------------------------

SUBNET provides health data about the clusters registered to the organization from the :guilabel:`Deployments` section.

The view shows the total size of the org's MinIO clusters with details for each cluster.

.. image:: /images/subnet/deployments-overview.png
   :width: 600px
   :alt: MinIO SUBNET displaying the deployments overview
   :align: center

Each of the organization's clusters display below the summary data.
Select a deployment row to view additional health details.

Deployment Health
~~~~~~~~~~~~~~~~~

The deployment's details include a summary of the deployment's configuration and the number of checks run and failed.
You can select :guilabel:`Upload` to add diagnostic health data obtained from the :mc-cmd:`mc support diagnostics` command or the MinIO Console's Support > Health page.

If you need support from MinIO Engineering, you can create a :guilabel:`New Issue` for the deployment.

.. image:: /images/subnet/SUBNET-deployment-health-summary.png
   :width: 600px
   :alt: MinIO SUBNET displaying health summary information for a myminio deployment
   :align: center

> SUBNET displays health checks for data points such as CPU, drives, memory, netowrk, and security.

Reviewer: ```suggestion
SUBNET displays health checks for data points such as CPU, drives, memory, network, and security.
```

Failed checks display first.
Checks with warnings display after failed checks.

> Finally, checks that pass display.

Reviewer: ```suggestion
Checks that pass display last.
```

Author: Parallel structure for the win!

=======
Support
=======

.. default-domain:: minio

.. contents:: Table of Contents
   :local:
   :depth: 1

Overview
--------

MinIO users have two options for support.

> #. Community support from the `public Slack channel <https://minio.slack.com>`_.

Reviewer: Should add something like "Community support is best-effort only and has no SLA or SLO"

Could do :abbr:`SLA (Service Level Agreement)` or :abbr:`SLO (Service Level Objective)` here. Not sure if I have the syntax right.

> #. A paid commercial license grants access to the MinIO Subscription Network, `SUBNET <https://subnet.min.io>`_.

Reviewer: Might want to check in we/ marketing to see if they have a preferred lead in to describing SUBNET.

Spitballing:

MinIO `SUBNET <https://subnet.min.io?ref=docs>` providers 24/7 Direct-to-engineering support w/ 24 hour or 1 hour SLA depending on subscription level.

   
   For current licensing levels and pricing, refer to the |SUBNET| page.

.. _minio-docs-subnet:

SUBNET
------

> SUBNET provides a support model that delivers 24/7/365 direct-to-engineer support thorugh a MinIO-builot portal that blends the best of Slack and Zendesk.

Reviewer: ```suggestion
SUBNET provides a support model that delivers 24/7/365 direct-to-engineer support through a MinIO-builot portal that blends the best of Slack and Zendesk.
```

Features of SUBNET include:

> - Security and architecture reviews

Reviewer: I think this is also subscription level based

- Access to Panic Button (depending on SLA)
- Secure communication channel to exchange logs and software binaries
- Unlimited seats for your team
- Unlimited issues

> Registering for SUBNET

Reviewer: Register your MinIO Deployment with SUBNET.

Otherwise this might read as how to actually sign up for SUBNET

   :align: center

Reviewing Health Data in SUBNET
-------------------------------

SUBNET provides health data about the clusters registered to the organization from the :guilabel:`Deployments` section.

The view shows the total size of the org's MinIO clusters with details for each cluster.

.. image:: /images/subnet/deployments-overview.png
   :width: 600px
   :alt: MinIO SUBNET displaying the deployments overview
   :align: center

Each of the organization's clusters display below the summary data.
Select a deployment row to view additional health details.

Deployment Health
~~~~~~~~~~~~~~~~~

The deployment's details include a summary of the deployment's configuration and the number of checks run and failed.
You can select :guilabel:`Upload` to add diagnostic health data obtained from the :mc-cmd:`mc support diagnostics` command or the MinIO Console's Support > Health page.

If you need support from MinIO Engineering, you can create a :guilabel:`New Issue` for the deployment.

.. image:: /images/subnet/SUBNET-deployment-health-summary.png
   :width: 600px
   :alt: MinIO SUBNET displaying health summary information for a myminio deployment
   :align: center

> When available, the following checks display for the deployment:

Reviewer: Thought - we have to maintain this list as it changes if we specify it above. I'm of a mind to just make this generic like "SUBNET displays health checks for data points such as CPU, drives, memory, network, and security."

- CPU Cores 
- CPU Flags
- CPU Match
- Drive Match
- Drive Status
- Drive Usage
- File System Match
- Health Report
- Memory Size
- Internode Latency (Roundtrip)
- MinIO Server Arguments
- MinIO Version
- NTP
- Server Process
- Swap Memory
- System Settings
- TLS config

Failed checks display first.
Checks with warnings display after failed checks.
Finally, checks that pass display.

Select any failed or warned checks to display the JSON output for additional details.
You can scroll vertically through the output for the selected check.

.. image:: /images/subnet/deployment-health-tls-config-fail.png
   :width: 600px
   :alt: MinIO SUBNET's health report for a deployment showing a failed Health Report with details expanded
   :align: center

Uploading Data to SUBNET
------------------------

If you registered the cluster with |SUBNET|, Performance and Inspection files can automatically upload to SUBNET.

> For clusters with an airgap, firewall, or otherwise disconnected from connecting to SUBNET directly, you can manually upload files.

Reviewer: "disconnected from connecting" is a bit heavy handed - 'otherwise blocked' is sufficient.

Reviewer: 'you can manually upload files to SUBNET health after logging in.'

> #. Generate the file(s) to upload from the :ref:`Console <minio-console>` or :ref:`Command Line <minio-client>`'s support functions

Reviewer: replace <minio-client> here with `mc support diagnostic` since afaik that's the only one that supports this functionality.

#. Sign in to `SUBNET <https://subnet.min.io>`_
#. Select :guilabel:`Deployments`
#. Select :guilabel:`Diagnostics`
#. Drag and drop the ``.gzip`` file(s) or browse to the file location to upload

Encrypting Data
~~~~~~~~~~~~~~~

> Data from the Inspect tool can be encrypted.

Reviewer: replace Inspect tool with `mc support inspect`
